cmd/frontend/internal/modelconfig/resolver.go

```diff
@@ -135,11 +135,59 @@ func (r *codyLLMConfigurationResolver) FastChatModelMaxTokens() (*int32, error)
 	return &maxTokens, nil
 }
 
+// Here Be Dragons (written July 30 2024)
+//
+// Cody clients currently rely on CodyLLMConfiguration, they shouldn't, they should
+// use the information provided by the ModelsService and the /.api/client-config -
+// both of which supersede this information. However, they use it today.
+//
+// Clients currently rely on this CodyLLMConfiguration.provider field ONLY to
+// determine which **autocomplete provider implementation** to use. That is, to
+// control context limits, prompting behavior, and other aspects of autocomplete.
+// This is not a great way to handle this (provider/model-specific behavior being
+// fundamentally tied together), but again, it's how it works today.
+//
+// The 'autocomplete provider name string' is determined by the following logic:
+// 1) If the server has the new `modelConfiguration` in their site config, then the
+// client will use the autocomplete `Model.provider` (provider ID, not name) field
+// as the string.
+// 2) Else `if (authStatus.configOverwrites?.provider)` - the string returned by this
+// function will be used.
```

Comment:
This is the intended syntax, right? i.e. that this second case from the Cody client code is the coming of this GraphQL endpoint? (Which is my understanding of where the `configOverwrites` data comes from.)

```suggestion
// 2) Else `if (authStatus.configOverwrites?.provider)` -- i.e. the string returned by this
// function -- will be used.
```

Reply:
That's correct.